blueprints/app/files/eslint.config.mjs

```diff
@@ -0,0 +1,69 @@
+<% if (!typescript) { %>import babelEslintParser from '@babel/eslint-parser';
+<% } %>import eslint from '@eslint/js';
+import pluginEmberRecommended from 'eslint-plugin-ember/configs/recommended';
+import pluginNode from 'eslint-plugin-n';
+import pluginPrettierRecommended from 'eslint-plugin-prettier/recommended';
+import pluginQunitRecommended from 'eslint-plugin-qunit/configs/recommended';
+<% if (typescript) { %>import typescriptEslint from 'typescript-eslint';
+<% } else { %>import globals from 'globals';
+<% } %>
+/** @type {import('eslint').Linter.FlatConfig[]} */
+export default [<% if (!typescript) { %>
+  // Babel:
+  {
+    languageOptions: {
+      globals: globals.browser,
+      parser: babelEslintParser,
+      parserOptions: {
+        babelOptions: {
+          plugins: [
+            [
+              '@babel/plugin-proposal-decorators',
+              { decoratorsBeforeExport: true },
+            ],
+          ],
+        },
+        ecmaVersion: 'latest',
+        requireConfigFile: false,
+        sourceType: 'module',
```

Comment:
`module` should be the default when using a flat config, so this can go?

Reply:
sourceType is determined by the package.json `type` field, so I think we still need this since we lie about what type of package our projects are (CJS)